cli/internal/syncer/reporters/plain_test.go

```diff
@@ -0,0 +1,25 @@
+package reporters_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/rudderlabs/rudder-iac/cli/internal/syncer/reporters"
+	"github.com/rudderlabs/rudder-iac/cli/internal/ui"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPlainSyncReporter(t *testing.T) {
+	r := &reporters.PlainSyncReporter{}
+
+	var buf bytes.Buffer
+	ui.SetWriter(&buf)
+	defer ui.ResetWriter()
+
+	confirmed, err := r.AskConfirmation()
+	assert.NoError(t, err)
+	assert.False(t, confirmed)
+
+	r.TaskCompleted("task1", "Task 1", nil)
+	assert.Equal(t, buf.String(), ui.Success("Task 1")+"\n")
```

Comment:
I think the `ui` abstraction is clean but given it's a singleton implementation it's hard to write concurrent tests where we might be setting / resetting writer's.

Not suggesting to change anything but generally globally variables although powerful can cause issues in concurrent cases.

Reply:
Yes, I'm facing similar issues, especially when I need to test the output, e.g when printing diffs.
I think a better approach would be to:
- Have the majority of ui functions return strings, so that they provide consistent formatting for error messages, tables, colors etc.
- Have the components that need to print something (e.g this reporter) expect a Writer as argument, which could default to stdout.

With those things I wouldn't even have to change the global writer in order to test. Let me see if it's a small lift to make those changes in this PR, at least in order to avoid this kind of global handling.